manifests/init.pp

```diff
@@ -48,94 +42,103 @@
+  $supported = lookup('openvmtools::supported', Boolean, 'first', false)
 
-  # Validate our booleans
-  validate_bool($with_desktop)
-  validate_bool($autoupgrade)
-  validate_bool($service_enable)
-  validate_bool($service_hasstatus)
-  validate_bool($supported)
 
-  case $ensure {
-    /(present)/: {
-      if $autoupgrade == true {
-        $package_ensure = 'latest'
-      } else {
-        $package_ensure = 'present'
-      }
-
-      if $service_ensure in [ running, stopped ] {
-        $service_ensure_real = $service_ensure
-      } else {
-        fail('service_ensure parameter must be running or stopped')
-      }
-    }
-    /(absent)/: {
-      $package_ensure = 'absent'
-      $service_ensure_real = 'stopped'
-    }
-    default: {
-      fail('ensure parameter must be present or absent')
+  if $ensure == 'present' {
+    $package_ensure = $autoupgrade ? {
+      true    => 'latest',
+      default => 'present',
     }
+    $service_ensure_real = $service_ensure
+  }
+  else {  # ensure == 'absent'
+    $package_ensure = 'absent'
+    $service_ensure_real = 'stopped'
   }
 
-  case $::virtual {
-    'vmware': {
-      if $supported {
-        package { $package_name :
-          ensure => $package_ensure,
-        }
-        if $with_desktop {
-          package { $desktop_package_name :
-            ensure => $package_ensure,
-          }
-        }
-
-        service { $service_name :
-          ensure    => $service_ensure_real,
-          enable    => $service_enable,
-          hasstatus => $service_hasstatus,
-          pattern   => $service_pattern,
-          require   => Package[$package_name],
+  if defined('$facts["virtual"]') && $facts['virtual'] == 'vmware' {
+    if $supported {
+      $packages = $with_desktop ? {
+        true    => $desktop_package_conflicts ? {
+          true    => [ $desktop_package_name ],
+          default => [ $package_name, $desktop_package_name ],
+        },
+        default => [ $package_name ],
+      }   
+      if $manage_epel {
+        include epel
+        Yumrepo['epel'] -> Package[$packages]
+      }
+      $vmware_uninstaller = defined('$facts["vmware_tools_uninstaller_path"]') ? {
+        true    => $facts['vmware_tools_uninstaller_path'],
+        default => null,
+      }
+      if $vmware_uninstaller =~ String {
+        $vmware_lib = $vmware_uninstaller.regex_replace('bin/vmware-uninstall-tools.pl','lib/vmware-tools')
+        exec { 'vmware-uninstall-tools':
+          command => "${vmware_uninstaller} && rm -rf ${vmware_lib}",
+          before  => Package['VMwareTools'],
         }
-      } else {
-        notice "Your operating system ${::operatingsystem} is unsupported and will not have the Open Virtual Machine Tools installed."
       }
+      package { 'VMwareTools':
+        ensure  => 'absent',
+        before  => Package[$packages],
+      }
+      ensure_packages($packages, { 'ensure' => $package_ensure })
+      ensure_resource('service', $service_name, {
```

Comment:
Why did you use `ensure_resource` here and not a simple `service` resource? I don't see a reason to use `ensure_resource()`.

Reply:
`$service_name` can be either a single service (a string) or multiple services (array of string).   This happens to exactly match the `ensure_resource` calling pattern.

On Linux, (not sure if Debian is an exception), openvmtools by default includes two services, one called `vmtoolsd` (though Debian renames it) and one called `vgauthd`.  This has been the case since December 2012, according to their changelog.

Reply:
Ah, I see. Well FYI then, both `package` and `service` accept  an Array of strings in the title.

```
$array_of_services = [ 'one', 'two' ]
service { $array_of_services:
  ensure => running,
}
```
```
$array_of_packages = [ 'one', 'two' ]
service { $array_of_packages:
  ensure => present,
}
```

Reply:
Gotcha; I had forgotten about that.  Reverted.